src/main/java/seedu/address/storage/InputCommandStorage.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.storage;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a storage for all user inputs keyed in the {@code CommandBox}.
+ */
+public class InputCommandStorage {
+
+    private static ArrayList<String> inputCommandList = new ArrayList<String>();
+    private static int currentPointer;
+
+    /**
+     * Constructor for InputCommandStorage. No inputs have been keyed in by user,
+     * CurrentPointer is set to 0, will traverse as keyCodes in
+     * {@code handleToggleQuery(KeyEvent event)} changes.
+     */
+    public InputCommandStorage() {
+        this.inputCommandList = inputCommandList;
```

Comment:
would be cleaner if this was `this.inputCommandList = new ArrayList<>();`

then line 10 can then be `private static List<String> inputCommandList`